packages/backend-common/src/service/lib/ServiceBuilderImpl.ts

```diff
@@ -120,20 +142,95 @@ export class ServiceBuilderImpl implements ServiceBuilder {
         reject(e);
       });
 
-      const server = stoppable(
-        app.listen(port, host, () => {
-          logger.info(`Listening on ${host}:${port}`);
-        }),
-        0,
-      );
+      let server: http.Server;
+
+      if (httpsSettings) {
+        logger.info('Initializing https server');
+
+        const credentials: { key: string; cert: string } = {
+          key: '',
+          cert: '',
+        };
+        const signingOptions: any = httpsSettings?.certificate;
+
+        if (signingOptions?.algorithm !== undefined) {
+          logger.info('Generating self-signed certificate with attributes');
+
+          const certificateAttributes: Array<any> = [];
+
+          Object.entries(signingOptions?.attributes).map(([name, value]) => {
+            certificateAttributes.push({
+              name: name,
+              value: value,
+            });
+          });
+
+          // TODO: Create a type def for selfsigned.
+          const signatures = require('selfsigned').generate(
+            certificateAttributes,
+            {
+              keySize: signingOptions?.size || 2048,
+              algorithm: signingOptions?.algorithm || 'sha256',
+              days: signingOptions?.days || 30,
+            },
+          );
+
+          logger.info(
+            'Bootstrapping key and cert from self-signed certificate',
```

Comment:
Was thinking logging before the `generate()` call, at this point we're already done waiting 😅

Reply:
I measured the time it take to generate a 2048 bit sha256 on my 8 year old home PC with an intel 7 chip. Its in the millisecond range so I dont think it matters to much and for me its more important to inform people that https was bootstrapped using whatever strategy they configured. With that said you cant have to many logging calls so I will add another one. :)

Reply:
Yeah saw you added both now so it's fine